base/compiler/abstractinterpretation.jl

```diff
@@ -537,8 +537,8 @@ function precise_container_type(interp::AbstractInterpreter, @nospecialize(itft)
         if _any(t -> !isa(t, DataType) || !(t <: Tuple) || !isknownlength(t), utis)
             return Any[Vararg{Any}], nothing
         end
-        result = Any[rewrap_unionall(p, tti0) for p in utis[1].parameters]
-        for t in utis[2:end]
+        result = Any[rewrap_unionall(p, tti0) for p in (utis[1]::DataType).parameters]
+        for t::DataType in utis[2:end]
```

Comment:
this adds `t = convert(t, DataType)::DataType`, which seems like it'd still trigger your tool?

Reply:
Actually I added these annotations manually, not by JET.
JET is unsound type checker and will give up reporting errors when a type is widen to `Any`.